File: labs/lab3/decrypt.py
import os
from Crypto.Cipher import AES
from Crypto.Util.Padding import unpad

INPUT_DIR = 'sdcard/DCIM/camera'
INPUT_FILE = 'sdcard/Download/text.txt'
KEY_FILE = 'sdcard/SECRETS/nothingsecrethere'
OUTPUT_DIR = 'decrypted_output'
OUTPUT_FILE = 'decrypted_output/text.txt'
IV = b'The BigIV\'s here'

def read_keys(keyfile_path):
    with open(keyfile_path, 'r') as f:
        return [line.strip() for line in f if line.strip()]

def try_decrypt(data, key):
    try:
        key_bytes = key.encode('utf-8')
        cipher = AES.new(key_bytes, AES.MODE_CBC, IV)
        decrypted = unpad(cipher.decrypt(data), AES.block_size)
        return decrypted
    except Exception:
        return None

def decrypt_directory(directory):
    if not os.path.exists(directory):
        os.makedirs(directory)

    keys = read_keys(KEY_FILE)
    files = [f for f in os.listdir(directory) if os.path.isfile(os.path.join(directory, f))]

    for file_name in files:
        file_path = os.path.join(directory, file_name)
        with open(file_path, 'rb') as f:
            encrypted_data = f.read()

        for key in keys:
            print(f"Trying key '{key}' on file: {file_name}")
            decrypted = try_decrypt(encrypted_data, key)
            if decrypted:
                output_path = os.path.join(OUTPUT_DIR, file_name)
                with open(output_path, 'wb') as out_f:
                    out_f.write(decrypted)
                print(f"✅ Decrypted '{file_name}' with key: {key}")
                break

def decrypt_file(file_path):
    if not os.path.exists(OUTPUT_DIR):
        os.makedirs(OUTPUT_DIR)

    keys = read_keys(KEY_FILE)
    file_name = os.path.basename(file_path)

    with open(file_path, 'rb') as f:
        encrypted_data = f.read()

    for key in keys:
        print(f"Trying key '{key}' on file: {file_name}")
        decrypted = try_decrypt(encrypted_data, key)
        if decrypted:
            output_path = os.path.join(OUTPUT_DIR, file_name)
            with open(output_path, 'wb') as out_f:
                out_f.write(decrypted)
            print(f"✅ Decrypted '{file_name}' with key: {key}")
            break

if __name__ == '__main__':
	decrypt_file(INPUT_FILE)
	decrypt_directory(INPUT_DIR)

File: labs/lab3/Decrypt.java
import java.io.*;
import java.nio.charset.StandardCharsets;
import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

class Decrypt {

	public static void decrypt(File file, String str) {
		byte[] decryptedData = new byte[0];
		try {
			FileInputStream fileInputStream = new FileInputStream(file);
			try {
				byte[] encryptedBytes = new byte[(int) file.length()];
				fileInputStream.read(encryptedBytes);
				SecretKeySpec secretKeySpec = new SecretKeySpec(str.getBytes(), "AES");
				Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
				cipher.init(
					Cipher.DECRYPT_MODE,
					secretKeySpec,
					new IvParameterSpec("The BigIV's here".getBytes(StandardCharsets.UTF_8))
				);
				decryptedData = cipher.doFinal(encryptedBytes);
				fileInputStream.close();
			} finally {}
		} catch (Exception e3) {
			e3 = e3;
		}

		File decrypted = new File("decrypted.txt");
		try {
			FileOutputStream fileOutputStream = new FileOutputStream(decrypted);
			try {
				fileOutputStream.write(decryptedData);
				fileOutputStream.close();
			} finally {}
		} catch (Exception e4) {
			e4 = e4;
		}
	}

	public static void main(String[] args) {
		File file = new File("sdcard/Download/text.txt");

		File keyFile = new File("sdcard/SECRETS/nothingsecrethere");

		try {
			BufferedReader reader = new BufferedReader(new FileReader(keyFile));
			String line;
			while ((line = reader.readLine()) != null) {
				try {
					decrypt(file, line);
				} catch (Exception e) {
					e = e;
				}
			}
			reader.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}
